src/simplewallet/simplewallet.cpp

```diff
@@ -612,129 +606,236 @@ namespace
+          if (!pw_prompter)
+          {
+            MERROR("Password needed without prompt function");
+            return boost::optional<tools::password_container>();
+          }
+          return pw_prompter("Daemon client password", verify);
+        }
+      );
+      if (!parsed)
+        return false;
+
+      daemon_username = std::move(parsed->username);
+      epee::wipeable_string daemon_pw = std::move(parsed->password).password();
+      daemon_password = std::string(daemon_pw.data(), daemon_pw.size());
     }
-    catch (const tools::error::not_enough_outs_to_mix& e)
+
+    return true;
+  }
+
+  bool get_daemon_init_settings(const boost::program_options::variables_map &vm,
+                                std::string &daemon_address,
+                                std::string &daemon_username,
+                                std::string &daemon_password,
+                                boost::optional<bool> &trusted_daemon,
+                                std::string &ssl_support,
+                                std::string &ssl_private_key_path,
+                                std::string &ssl_certificate_path,
+                                std::string &ssl_ca_file_path,
+                                std::vector<std::string> &ssl_allowed_fingerprints_str,
+                                bool &ssl_allow_any_cert,
+                                std::string &proxy)
+  {
+    if (!get_daemon_address(vm, daemon_address, daemon_username, daemon_password))
+      return false;
+
+    ssl_support          = command_line::get_arg(vm, arg_daemon_ssl);
+    ssl_private_key_path = command_line::get_arg(vm, arg_daemon_ssl_private_key);
+    ssl_certificate_path = command_line::get_arg(vm, arg_daemon_ssl_certificate);
+    ssl_ca_file_path     = command_line::get_arg(vm, arg_daemon_ssl_ca_certificates);
+    ssl_allowed_fingerprints_str = command_line::get_arg(vm, arg_daemon_ssl_allowed_fingerprints);
+    ssl_allow_any_cert   = command_line::get_arg(vm, arg_daemon_ssl_allow_any_cert);
+
+    const bool use_proxy = command_line::has_arg(vm, arg_proxy);
+
+    // user specified CA file or fingeprints implies enabled SSL by default
+    epee::net_utils::ssl_options_t ssl_options = epee::net_utils::ssl_support_t::e_ssl_support_enabled;
+    if (ssl_allow_any_cert)
+      ssl_options.verification = epee::net_utils::ssl_verification_t::none;
+    else if (!ssl_ca_file_path.empty() || !ssl_allowed_fingerprints_str.empty())
     {
-      auto writer = fail_msg_writer();
-      writer << sw::tr("not enough outputs for specified ring size") << " = " << (e.mixin_count() + 1) << ":";
-      for (std::pair<uint64_t, uint64_t> outs_for_amount : e.scanty_outs())
+      std::vector<std::vector<uint8_t>> ssl_allowed_fingerprints{ ssl_allowed_fingerprints_str.size() };
+      std::transform(ssl_allowed_fingerprints_str.begin(), ssl_allowed_fingerprints_str.end(), ssl_allowed_fingerprints.begin(), epee::from_hex_locale::to_vector);
```

Comment:
`epee::from_hex_locale::to_vector` can throw exceptions if something is not ok with the parameters. Can this be a problem?